### PR TITLE
[JENKINS-42446] Triggers should be started on job creation

### PR DESCRIPTION
**Pull request by scoheb**
[JENKINS-42446](https://issues.jenkins-ci.org/browse/JENKINS-42446)

### CODE REVIEW

**Comment by scoheb** [comment]
@jenkinsci/code-reviewers

File: src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java [hunk 1]
```diff
@@ -196,6 +196,8 @@ public void setDefinition(FlowDefinition definition) {
         } else {
             quietPeriod = null;
         }
+        getTriggersJobProperty().stopTriggers();
+        getTriggersJobProperty().startTriggers(Items.currentlyUpdatingByXml());
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java]: Not really the right place to do this. `setOwner` seems more appropriate, but this was dropped in 09adb56241cbcac3baf2625f063af8da7ba87bee; perhaps @abayer recalls why.

File: src/test/java/org/jenkinsci/plugins/workflow/job/properties/PipelineTriggersJobPropertyTest.java [hunk 2]
```diff
@@ -93,10 +94,10 @@ public void submitCallsStartTrue() throws Exception {
         t = (MockTrigger) p.getTriggers().get(t.getDescriptor());
         assertNotNull(t);
         // The  first "null, false" is due to the p.addTrigger(t) call and won't apply in the real world.
-        assertEquals("[null, false, null, true]", MockTrigger.startsAndStops.toString());
+        assertEquals("[null, false, null, true, null, false]", MockTrigger.startsAndStops.toString());
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/job/properties/PipelineTriggersJobPropertyTest.java]: So it seems that we are now calling `stop()` and `start(false)` gratuitously after a form resubmission.

File: src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java (hunk 1 above)
Comment (abayer) [inline src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java]: Not a clue. The trigger stop/start stuff is black magic - I basically just worry about whether the tests pass.

**Comment by daniel-beck** [comment]
Does this break `isHoldOffBuildUntilSave()`?

**Comment by jglick** [comment]
@abayer so what about the redundant stop/start calls? Does that not look wrong?

File: src/test/java/org/jenkinsci/plugins/workflow/job/properties/PipelineTriggersJobPropertyTest.java (hunk 2 above)
Comment (abayer) [reply src/test/java/org/jenkinsci/plugins/workflow/job/properties/PipelineTriggersJobPropertyTest.java]: No one actually understands why or when we're calling those, honestly.

**Pull request merged**